src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java

```diff
@@ -1255,6 +1285,19 @@ protected void wipeOfTestResources(
         }
     }
 
+    protected void validateDocCountAndEmbedding(String indexName, int expectedDocCount, String docId, final String embeddingField) {
+        int count = getDocCount(indexName);
+        assertEquals(expectedDocCount, count);
+        Map<String, Object> document = getDocById(indexName, docId);
+        Object documentSource = document.get("_source");
+        assertTrue(documentSource instanceof Map);
+        @SuppressWarnings("unchecked")
+        Map<String, Object> documentSourceMap = (Map<String, Object>) documentSource;
+        assertTrue(documentSourceMap.containsKey(embeddingField));
+        Object ingestOutputs = documentSourceMap.get(embeddingField);
+        assertTrue(ingestOutputs instanceof Map);
```

Comment:
Why are we asserting that ingestOutputs is an instance of Map? I think dense embedding should be a vector, which is a list of numbers.

Reply:
We use sparse encoding to test batch ingestion so verify map type here.

Reply:
This function is from the base class, so we cannot assume this function will only be used for batch ingestion.

Reply:
You can verify `instanceof Map` from batch ingestion IT. Also for sparse encoding, a `Map` type is not enough. We need to verify type `Map<String, Double>`.